lab_2_retrieval_w_bm25/main.py

```diff
@@ -77,6 +110,16 @@ def calculate_idf(vocab: list[str], documents: list[list[str]]) -> dict[str, flo
 
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(vocab, list) or not all(isinstance(item, str) for item in vocab) or not isinstance(documents, list) or not all(isinstance(docum, list) for docum in documents) or not all(isinstance(item, str) for docum in documents for item in docum) or not vocab or not documents:
+        return None
+    idf_result = {}
+    for word in vocab:
+        words_count = 0
+        for doc in documents:
+            if word in doc:
+                words_count += 1
```

Comment:
Do you think it is possible to use sum() here to calculate words_count? If so, try to implement it

Reply:
Done